internal/app/frontend/frontend_service.go

```diff
@@ -374,16 +384,32 @@ func (s *frontendService) AcknowledgeBackfill(ctx context.Context, req *pb.Ackno
 	}
 
 	if len(associatedTickets) != 0 {
-		_, _, err = s.store.UpdateAssignments(ctx, &pb.AssignTicketsRequest{
+		_, tickets, err := s.store.UpdateAssignments(ctx, &pb.AssignTicketsRequest{
 			Assignments: []*pb.AssignmentGroup{&pb.AssignmentGroup{TicketIds: associatedTickets, Assignment: req.GetAssignment()}},
 		})
 		if err != nil {
 			return nil, err
 		}
+
+		// all assigned tickets should be removed from the associatedTickets
+		// associatedTickets - tickets
+		unassignedTickets := []string{}
+		assigned := make(map[string]struct{})
```

Comment:
Since later you are iterating for every ticket, setting `make(map[string]struct{}, len(tickets))` could be an improvement